src/codegen/tests/statement_codegen_test.rs

```diff
@@ -154,3 +154,24 @@ fn function_result_assignment_on_aliased_string() {
 
     insta::assert_snapshot!(result);
 }
+
+#[test]
+fn floating_point_type_casting() {
+    let result = codegen(
+        r#"
+        FUNCTION fn : DINT
+            VAR
+                a : REAL  :=       7 / 2; // => 3.0 (because we do a integer division first and only then cast the result)
```

Comment:
I know these cases are representative, but could you add body entries as well?
Also a correctness test would be good here.